chalk-solve/src/clauses/builtin_traits/sized.rs

```diff
@@ -43,3 +34,53 @@ pub fn add_sized_program_clauses<I: Interner>(
     };
     builder.push_clause(trait_ref.clone(), iter::once(last_field_sized_goal));
 }
+
+fn handle_tuple<I: Interner>(
+    db: &dyn RustIrDatabase<I>,
+    builder: &mut ClauseBuilder<'_, I>,
+    trait_ref: &TraitRef<I>,
+    arity: usize,
+    substitution: &Substitution<I>,
+) {
+    // Empty tuples are always Sized
+    if arity == 0 {
+        builder.push_fact(trait_ref.clone());
+        return;
+    }
+
+    let interner = db.interner();
+
+    // To check if a tuple is Sized, we only have to look at its last element.
+    // This is because the WF checks for tuples require that all the other elements must be Sized.
+    let last_elem_sized_goal = TraitRef {
+        trait_id: trait_ref.trait_id,
+        substitution: Substitution::from1(interner, substitution.iter(interner).last().unwrap()),
+    };
+    builder.push_clause(trait_ref.clone(), iter::once(last_elem_sized_goal));
+}
+
+pub fn add_sized_program_clauses<I: Interner>(
+    db: &dyn RustIrDatabase<I>,
+    builder: &mut ClauseBuilder<'_, I>,
+    trait_ref: &TraitRef<I>,
+    ty: &TyData<I>,
+) {
+    match ty {
+        TyData::Apply(ApplicationTy {
+            name: TypeName::Struct(struct_id),
+            substitution,
+        }) => handle_struct(db, builder, trait_ref, *struct_id, substitution),
+        // Scalars are always Sized
+        TyData::Apply(ApplicationTy {
+            name: TypeName::Scalar(_),
+            ..
+        }) => builder.push_fact(trait_ref.clone()),
+        TyData::Apply(ApplicationTy {
+            name: TypeName::Tuple(arity),
+            substitution,
+        }) => handle_tuple(db, builder, trait_ref, *arity, substitution),
+        // TODO(areredify)
+        // when #368 lands, extend this to handle everything accordingly
+        _ => return,
+    }
```

Comment:
I feel like this match is a little too verbose because we are decomposing the same variant in each of them. Also, I just realized that `TyData::Function` should be Sized too

Reply:
I've added a nested match statement for `TypeName`. I'd rather handle `Function` in another PR if possible.

Reply:
Why? Functions are always Sized, you just need to `push_fact` or whatever it's called

Reply:
Because i don't know how to write tests for it

Reply:
```
goal {
   fn(()): Sized
}
```
I believe this is syntax for a function that takes a unit type as input

Reply:
Done!